### PR TITLE
Update for 0.24 - Round 2

### PR DESCRIPTION
**Pull request by reshmabidikar**
Updates to slate docs for the following issues:

- #[1658](https://github.com/killbill/killbill/issues/1658)
- #[1651](https://github.com/killbill/killbill/issues/1651)
- #[1627](https://github.com/killbill/killbill/issues/1627)

### CODE REVIEW

File: source/includes/_invoice.md [hunk 1]
```diff
@@ -2658,6 +2658,357 @@ Note that for `SUBSCRIPTION_ACTION`, there are 2 dates to take into account:
 
 If successful, returns a status code of 200 and an invoice resource object.
 
+## Invoice Groups
+
+An invoice group is a group of invoices. An invoice group plugin can be used to split an invoice into multiple invoices. Such split invoices belong to a single invoice group. 
+
+Note that Kill Bill does not provide any reference implementation for an invoice group plugin. So anyone wishing to use this feature would need to create an invoice plugin and implement the [InvoicePluginApi#getInvoiceGrouping](https://github.com/killbill/killbill-plugin-api/blob/20e10d463b67b0434e1b9f4dd62ed34ebe84bd9b/invoice/src/main/java/org/killbill/billing/invoice/plugin/api/InvoicePluginApi.java#L41) method. We provide a [demo plugin](https://github.com/killbill/killbill-invgrp-demo) that demonstrates how an invoice group plugin can be implemented. When such a plugin is configured, an [invoice group run](https://killbill.github.io/slate/#invoice-trigger-an-invoice-group-run) would result in the generation of multiple invoices.
```
Comment (sbrossie) [inline source/includes/_invoice.md:2665]: @reshmabidikar I had a discussion with @pierre , and we thought we would rename the [killbill-invgrp-demo](https://github.com/killbill/killbill-invgrp-demo) to something like `killbill-invgrp` - after all this is a fully functional plugin. In which, we could then say that such a plugin can be used as a reference implementation.

I don't suggest any changes for now, but if you could track it, then we could update plugin name and doc together.

File: source/includes/_invoice.md [hunk 2]
```diff
@@ -2658,6 +2658,357 @@ Note that for `SUBSCRIPTION_ACTION`, there are 2 dates to take into account:
...
+```shell
+# Subset of headers returned when specifying -v curl option
+< HTTP/1.1 201
+< Content-Type: application/json
+< Content-Length: 0
+< Location: http://127.0.0.1:8080/1.0/kb/invoices/60f5af97-9457-4b53-9eec-aec1de96a66f/group
+```
+
+```java
+TODO
```
Comment (sbrossie) [inline source/includes/_invoice.md:2721]: If you include the request for `java`, then we probably need the response as well.

**Review by sbrossie — APPROVED** [review]

File: source/includes/_invoice.md (hunk 1 above)
Comment (reshmabidikar) [reply source/includes/_invoice.md:2665]: Acknowledged, I've filed https://github.com/killbill/technical-support/issues/89 to track this.

File: source/includes/_invoice.md (hunk 2 above)
Comment (reshmabidikar) [reply source/includes/_invoice.md]: Fixed. I think there is actually an issue while retrieving the response when this endpoint is executed via the Java client. I had filed https://github.com/killbill/technical-support/issues/87 to investigate this further. For now, I've copied over the response from the `Retrieve Invoice Group` endpoint.

**Pull request merged**